src/editor/component_panels/variables_panel.cpp

```diff
@@ -133,9 +134,18 @@ void OrchestratorScriptVariablesComponentPanel::_handle_context_menu(int p_id)
         case CM_REMOVE_VARIABLE:
             _confirm_removal(_tree->get_selected());
             break;
+        case CM_DUPLICATE_VARIABLE:
+            _duplicate_variable(_tree->get_selected());
     }
 }
 
+void OrchestratorScriptVariablesComponentPanel::_duplicate_variable(TreeItem* p_item)
+{
+    Ref<OScriptVariable> duplicate = _orchestration->duplicate_variable(_get_tree_item_name(p_item));
+    update();
+    _find_child_and_activate(duplicate->get_variable_name());
```

Comment:
I would guard these two lines with a `if (duplicate.is_valid())` check for good measure.